### PR TITLE
ScalazPlugin

### PR DESCRIPTION
**Pull request by rossabaker**
This is a test run for a Scalaz plugin I've been sprinkling about various repositories and preparing to launch as an open source project.  This is its latest iteration.

A future commit will remove this in favor of a centralized dependency, and we'll be able to use it in repos like http4s-argonaut61.

Most of these changes will be stripped when merging to cats.

### CODE REVIEW

**Comment by rossabaker** [comment]
The `-scalaz-x.y` versioning is what we are using on [Verizon OSS](https://github.com/Verizon).  It also has the positive property of sorting qualifiers based in current order of maturity (`scalaz-7.2` > `scalaz-7.1` > `cats`), and http4s' long term desire (drop `cats` suffix once it's the recommended default, and it moves to the front of the list).  It's also compatible with semver if you look at it just right.

It has the negative of introducing a new convention in http4s hopefully not too many releases before scalaz support is dropped for cats.  I am open to preserving the status quo, which is a small change here.

File: project/Http4sPlugin.scala [hunk 1]
```diff
@@ -115,15 +104,15 @@ object Http4sPlugin extends AutoPlugin {
...
+  def scalaReflect(so: String, sv: String) = so              % "scala-reflect"           % sv
   lazy val scalaXml            = "org.scala-lang.modules"   %% "scala-xml"               % "1.0.6"
-  def scalazCore(version: String)               = "org.scalaz"           %% "scalaz-core"               % version
-  def scalazScalacheckBinding(version: String)  = "org.scalaz"           %% "scalaz-scalacheck-binding" % version
-  def specs2Core(scalazVersion: String)         = "org.specs2"           %% "specs2-core"               % specs2Version(scalazVersion)
-  def specs2MatcherExtra(scalazVersion: String) = "org.specs2"           %% "specs2-matcher-extra"      % specs2Core(scalazVersion).revision
-  def specs2Scalacheck(scalazVersion: String)   = "org.specs2"           %% "specs2-scalacheck"         % specs2Core(scalazVersion).revision
-  def scalazStream(scalazVersion: String)       = "org.scalaz.stream"    %% "scalaz-stream"             % scalazCrossBuild("0.8.6", scalazVersion)
+  def scalazCore(szv: String)  = "org.scalaz"               %% "scalaz-core"             % szv
+  def scalazScalacheckBinding(szv: String)  = "org.scalaz"  %% "scalaz-scalacheck-binding" % szv
```
Comment (ChristopherDavenport) [inline project/Http4sPlugin.scala]: Nitpicky, this rows version is now out of alignment and will drive me mad.

Comment (rossabaker) [inline project/Http4sPlugin.scala]: Are you referring to the one extra space after scalazScalacheckBinding?  I was trying to align everything to where most of the dependencies were, in the soonest column possible.

I almost realigned the whole table.  I'm tempted.

Comment (cquiroz) [inline project/Http4sPlugin.scala]: 👍  I agree with @ChristopherDavenport

**Comment by rossabaker** [comment]
I decided not to rock the boat on the dependency suffixing.

**Pull request merged**